File: app/Http/Controllers/DashboardController.php
<?php

namespace App\Http\Controllers;

use App\Services\BatchService;
use Illuminate\View\View;

class DashboardController extends Controller
{
    protected BatchService $batchService;

    public function __construct(BatchService $batchService)
    {
        $this->batchService = $batchService;
    }

    public function index(): View
    {
        $stats = $this->batchService->getDashboardStatistics();

        return view('dashboard', [
            'stats' => $stats,
            'seedTypes' => $this->batchService->getSeedTypes(),
        ]);
    }
}

File: Controllers/DashboardController.cs
using MeetingRoomEquipment.Services;
using MeetingRoomEquipment.ViewModels;
using Microsoft.AspNetCore.Mvc;

namespace MeetingRoomEquipment.Controllers;

public class DashboardController : Controller
{
    private readonly IStatisticsService _stats;
    private readonly ICurrentUserService _user;

    public DashboardController(IStatisticsService stats, ICurrentUserService user)
    {
        _stats = stats;
        _user = user;
    }

    public async Task<IActionResult> Index()
    {
        var vm = await _stats.GetDashboardAsync();
        ViewData["CurrentUser"] = _user.UserName;
        ViewData["CurrentRole"] = _user.Role;
        return View(vm);
    }
}
